.github/workflows/ci-sglang-benchmark.yml

```
@@ -186,18 +176,65 @@ jobs:
 
       - name: Run SGLang Benchmark Tests
         run: |
-          pytest -v app_tests/benchmark_tests/llm/sglang_benchmarks/sglang_benchmark_test.py --port 30000 --log-cli-level=INFO --html=out/llm/sglang/index.html
+          pytest -v app_tests/benchmark_tests/llm/sglang_benchmarks/sglang_benchmark_test.py --port 30000 --log-cli-level=INFO --html=sglang_index.html --self-contained-html
 
       - name: Stop sglang-server
         run: docker stop sglang-server || true # Stop container if it's running
 
+      # Deleting image after run due to large disk space requirement (83 GB)
       - name: Cleanup SGLang Image
         run: docker image rm lmsysorg/sglang:v0.3.5.post1-rocm620
 
+      - name: Upload pytest report
+        uses: actions/upload-artifact@b4b15b8c7c6ac21ea08fcf65892d2ee8f75cf882
+        with:
+          name: sglang_benchmark
+          path: sglang_index.html
+
+  merge_and_upload_reports:
+    if: ${{ github.repository_owner == 'nod-ai' || github.event_name != 'schedule' }}
+    name: "Merge and upload benchmark reports"
+    needs: [benchmark_shortfin, benchmark_sglang]
+    strategy:
+      matrix:
+        version: [3.11]
+      fail-fast: false
+    runs-on: ubuntu-24.04
+    defaults:
+      run:
+        shell: bash
+    steps:
+      - name: "Setting up Python"
+        id: setup_python
+        uses: actions/setup-python@0b93645e9fea7318ecaed2b359559ac225c90a2b # v5.3.0
+        with:
+          python-version: ${{matrix.version}}
+
+      - name: Install pytest-html-merger
+        run: pip install pytest-html-merger
+
+      - name: Download shortfin report
+        uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16
+        with:
+          name: shortfin_benchmark
+          path: reports
+
+      - name: Download sglang report
+        uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16
+        with:
+          name: sglang_benchmark
+          path: reports
+
+      - name: Create merged report directory
+        run: mkdir merged_reports
+
+      - name: Merge html reports
+        run: pytest_html_merger -i reports -o merged_reports/index.html
```

nit: I'd personally merge these two steps. I don't see much value in having a step for just `mkdir`

```suggestion
      - name: Merge html reports
        run: |
          mkdir merged_reports
          pytest_html_merger -i reports -o merged_reports/index.html
```

---

Updated